src/pages/LoginForms/ControlledComponent.jsx

```diff
@@ -0,0 +1,115 @@
+import React, { useState } from "react";
+import { useReducer } from "react";
+
+export default function ControlledComponent() {
+  const [state, dispatch] = useReducer(reducer, {
+    login: "",
+    password: "",
+    validLogin: true,
+    validPassword: true,
+  });
+
+  function reducer(state, action) {
+    switch (action.type) {
+      case "setLogin":
```

Comment:
"setLogin" and other action types should be moved in a const. E.g. 
```
const loginFormActionTypes = {
    SET_LOGIN: 'SET_LOGIN',
    ....
}
```
Later you should refer to each action type as loginFormActionTypes.SET_LOGIN.

Reply:
fixed